src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthHelper.java

```diff
@@ -0,0 +1,184 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.auth.connector.okta;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.nike.backstopper.exception.ApiException;
+import com.nike.cerberus.error.DefaultApiError;
+import com.okta.sdk.clients.AuthApiClient;
+import com.okta.sdk.clients.UserApiClient;
+import com.okta.sdk.models.auth.AuthResult;
+import com.okta.sdk.models.factors.Factor;
+import com.okta.sdk.models.usergroups.UserGroup;
+import org.apache.commons.lang3.text.WordUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helper methods to authenticate with Okta.
+ */
+public class OktaAuthHelper {
+
+    public static final String AUTHENTICATION_MFA_REQUIRED_STATUS = "MFA_REQUIRED";
+
+    public static final String AUTHENTICATION_SUCCESS_STATUS = "SUCCESS";
+
+    private static final ImmutableMap<String, String> MFA_FACTOR_NAMES = ImmutableMap.of(
+            "google", "Google Authenticator",
+            "okta"  , "Okta Verify");
+
+    private final ObjectMapper objectMapper;
+
+    private final AuthApiClient authClient;
+
+    private final UserApiClient userApiClient;
+
+    @Inject
+    public OktaAuthHelper(final AuthApiClient authApiClient, final UserApiClient userApiClient) {
+
+        this.objectMapper = new ObjectMapper();
+        this.objectMapper.findAndRegisterModules();
+        this.objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+        this.objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        this.objectMapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        this.objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        this.objectMapper.enable(SerializationFeature.WRITE_EMPTY_JSON_ARRAYS);
+        this.objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        this.authClient = authApiClient;
+        this.userApiClient = userApiClient;
+    }
+
+    /**
+     * Request to get user group data by the user's ID.
+     *
+     * @param userId User ID
+     * @return User groups
+     */
+    protected List<UserGroup> getUserGroups(final String userId) {
+
+        try {
+            return this.userApiClient.getUserGroups(userId);
+        } catch (IOException ioe) {
+            final String msg = String.format("failed to get user groups for reason: %s", ioe.getMessage());
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.GENERIC_BAD_REQUEST)
```

Comment:
This doesn't seem to be the appropriate response.

See https://github.com/Nike-Inc/cerberus-management-service/blob/master/src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java#L241